sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsAsyncClient.java

```diff
@@ -143,17 +143,17 @@ public DigitalTwinsServiceVersion getServiceVersion() {
      * Creates a digital twin.
      *
      * @param digitalTwinId The Id of the digital twin.
-     * @param digitalTwin The application/json digital twin to create.
-     * @param clazz The model class to deserialize the response with.
-     * @param <T> The generic type to deserialize the digital twin with.
+     * @param digitalTwin The application/json object representing the digital twin to create.
+     * @param clazz The model class to serialize the request with and deserialize the response with.
+     * @param <T> The generic type to serialize the request with and deserialize the response with.
      * @return A {@link DigitalTwinsResponse} containing the deserialized application/json object representing the digital twin created.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public <T> Mono<DigitalTwinsResponse<T>> createDigitalTwinWithResponse(String digitalTwinId, Object digitalTwin, Class<T> clazz) {
```

Comment:
@abhipsaMisra I believe you were in favor of taking an Object here rather than a T, so I wanted to discuss this again. In our e2e tests, this API was being called such that the digitalTwin argument was a String and the return type was of type T (BasicDigitalTwin). Is that expected? My understanding was that we wanted to take and return the same type here.

Reply:
Same here and below with createRelationship APIs

Reply:
This is my thought:
My intention was more around flexibility with this. The `T` that we accept as input isn't being used anywhere in the request (all classes are objects, so we pass it in as-is). As for the serialization part of it, if the input is a valid json string, the PL will take it as-is. If not, then we will serialize it before sending it on the wire.

As for accepting `T` vs `Object`, since the `T` class info is unused in the request, it made more sense to me to have it as an object. In that case, I don't need to necessarily deserialize my request into a `BasicDigitalTwin` type, just so I can accept the response as a `BasicDigitalTwin`.
Again, even if we do mark the input as `T`, and I still pass in a json string, it will also work as expected; but now the public API is stating a contract that for you to be able to inspect a response as a `BasicDigitalTwin`, you need to pass in your input as a `BasicDigitalTwin` -> even though this is not a requirement.

Reply:
> Again, even if we do mark the input as T, and I still pass in a json string, it will also work as expected


I don't follow this part. If the input parameter is T, and you set T to be BasicDigitalTwin, then you cannot pass in a json string. You'll get a compilation error since the json string is not of type BasicDigitalTwin.

> As for accepting T vs Object, since the T class info is unused in the request, it made more sense to me to have it as an object.


I understand that we don't need the parameter to be of type T, but I don't want customers using this API in ways were the return type and passed in type are different. That sounds like a promise that we won't be able to keep given how generic Object is. I'd rather keep it simple and restful and force the users to give and receive the same type here.

> ... but now the public API is stating a contract that for you to be able to inspect a response as a BasicDigitalTwin, you need to pass in your input as a BasicDigitalTwin -> even though this is not a requirement.


I think this is a good expectation to place on the user, though. The whole point of the service is that customers are responsible for defining these types. If a customer is using this T flavor of our create APIs, then they should know the type, so why would they want to send it as one type and have us return the same object as a different type?

Reply:
I think using the Type T instead of object is the clear expectation from the user ... that you pass in an object of type T that you expect to get back ...
By just providing an untyped object, we are basically allowing them to pass in a K type and expect a T as the result. This can also mislead users and allow for a misuse in the API.
I think if we don't have a response type to deserialize the response into, it makes sense to take in an Object. but if we are doing a deserialization into a type, we should also take the input as the same type.

Reply:
I said type like 12 times.

Reply:
> So why would they want to send it as one type and have us return the same object as a different type

This is something we might run into since we expose these different overload, but I agree, we can set the expectation that your input and output should be the same type and see the feedback. I an not opposed to having the input as `T`, it was more of a flexibility thing, but there is no need to solve a problem that doesn't exist.

Go for `T`!